### ISSUE TITLE
[colmap] Build error

### ISSUE THREAD

**Issue by kwin120** [microsoft/vcpkg#30157]
**Host Environment**

- OS: x64-windows
- Compiler: MSVC 19.34.31942.0
- vcpkg-tool version: 2023-03-01-538539c4e364dd6f14f5e36f6704144254845159
- vcpkg-scripts version: dd1b57240 2023-03-09 (4 days ago)

**To Reproduce**

`vcpkg install colmap[cuda,tests]:x64-windows`

**Failure logs**

```
-- Using cached colmap-colmap-29a1e3642a3b00734a52b21e597ea4d576485fe6.tar.gz.
-- Cleaning sources at G:/GitHub/vcpkg/buildtrees/colmap/src/d576485fe6-7dd36362d1.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source G:/GitHub/vcpkg/downloads/colmap-colmap-29a1e3642a3b00734a52b21e597ea4d576485fe6.tar.gz
-- Applying patch fix-dependencies.patch
-- Using source at G:/GitHub/vcpkg/buildtrees/colmap/src/d576485fe6-7dd36362d1.clean
-- Found external ninja('1.11.0').
-- Configuring x64-windows-dbg
-- Configuring x64-windows-rel
-- Building x64-windows-dbg
-- Building x64-windows-rel
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: "C:/Program Files/CMake/bin/cmake.exe" --build . --config Release --target install -- -v -j13
    Working Directory: G:/GitHub/vcpkg/buildtrees/colmap/x64-windows-rel
    See logs for more information:
      G:\GitHub\vcpkg\buildtrees\colmap\install-x64-windows-rel-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  ports/colmap/portfile.cmake:59 (vcpkg_cmake_install)
  scripts/ports.cmake:147 (include)



```
<details><summary>G:\GitHub\vcpkg\buildtrees\colmap\install-x64-windows-rel-out.log</summary>

```
[1/342] "D:\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.34.31933\bin\Hostx64\x64\cl.exe"   /TP -DCGAL_ENABLED -DCUDA_ENABLED -DGLOG_NO_ABBREVIATED_SEVERITIES -DGL_GLEXT_PROTOTYPES -DGUI_ENABLED -DOPENGL_ENABLED -DOPENMP_ENABLED -DRELEASE -IG:\GitHub\vcpkg\buildtrees\colmap\src\d576485fe6-7dd36362d1.clean\lib -IG:\GitHub\vcpkg\buildtrees\colmap\src\d576485fe6-7dd36362d1.clean\src -IG:\GitHub\vcpkg\installed\x64-windows\include -IG:\GitHub\vcpkg\installed\x64-windows\include\eigen3 -IG:\GitHub\vcpkg\installed\x64-windows\include\qt5 -IG:\GitHub\vcpkg\installed\x64-windows\include\qt5\QtCore -IG:\GitHub\vcpkg\installed\x64-windows\tools\qt5\mkspecs\win32-msvc -IG:\GitHub\vcpkg\installed\x64-windows\include\qt5\QtOpenGL -IG:\GitHub\vcpkg\installed\x64-windows\include\qt5\QtWidgets -IG:\GitHub\vcpkg\installed\x64-windows\include\qt5\QtGui /nologo /DWIN32 /D_WINDOWS  /utf-8 /GR /EHsc /MP  /EHsc /MP -openmp /W0 /MD /O2 /Oi /Gy /DNDEBUG /Z7 /showIncludes /Folib\PoissonRecon\CMakeFiles\poisson_recon.dir\CmdLineParser.cpp.obj /Fdlib\PoissonRecon\CMakeFiles\poisson_recon.dir\poisson_recon.pdb /FS -c G:\GitHub\vcpkg\buildtrees\colmap\src\d576485fe6-7dd36362d1.clean\lib\PoissonRecon\CmdLineParser.cpp
[2/342] "D:\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.34.31933\bin\Hostx64\x64\cl.exe"   /TP -DCGAL_ENABLED -DCUDA_ENABLED -DGLOG_NO_ABBREVIATED_SEVERITIES -DGL_GLEXT_PROTOTYPES -DGUI_ENABLED -DOPENGL_ENABLED -DOPENMP_ENABLED -DRELEASE -IG:\GitHub\vcpkg\buildtrees\colmap\src\d576485fe6-7dd36362d1.clean\lib -IG:\GitHub\vcpkg\buildtrees\colmap\src\d576485fe6-7dd36362d1.clean\src -IG:\GitHub\vcpkg\installed\x64-windows\include -IG:\GitHub\vcpkg\installed\x64-windows\include\eigen3 -IG:\GitHub\vcpkg\installed\x64-windows\include\qt5 -IG:\GitHub\vcpkg\installed\x64-windows\include\qt5\QtCore -IG:\GitHub\vcpkg\installed\x64-windows\tools\qt5\mkspecs\win32-msvc -IG:\GitHub\vcpkg\installed\x64-windows\include\qt5\QtOpenGL -IG:\GitHub\vcpkg\installed\x64-windows\include\qt5\QtWidgets -IG:\GitHub\vcpkg\installed\x64-windows\include\qt5\QtGui /nologo /DWIN32 /D_WINDOWS  /utf-8 /GR /EHsc /MP  /EHsc /MP -openmp /W0 /MD /O2 /Oi /Gy /DNDEBUG /Z7 /showIncludes /Folib\PoissonRecon\CMakeFiles\poisson_recon.dir\Geometry.cpp.obj /Fdlib\PoissonRecon\CMakeFiles\poisson_recon.dir\poisson_recon.pdb /FS -c G:\GitHub\vcpkg\buildtrees\colmap\src\d576485fe6-7dd36362d1.clean\lib\PoissonRecon\Geometry.cpp
[3/342] "D:\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.34.31933\bin\Hostx64\x64\cl.exe"   /TP -DCGAL_ENABLED -DCUDA_ENABLED -DGLOG_NO_ABBREVIATED_SEVERITIES -DGL_GLEXT_PROTOTYPES -DGUI_ENABLED -DOPENGL_ENABLED -DOPENMP_ENABLED -DRELEASE -IG:\GitHub\vcpkg\buildtrees\colmap\src\d576485fe6-7dd36362d1.clean\lib -IG:\GitHub\vcpkg\buildtrees\colmap\src\d576485fe6-7dd36362d1.clean\src -IG:\GitHub\vcpkg\installed\x64-windows\include -IG:\GitHub\vcpkg\installed\x64-windows\include\eigen3 -IG:\GitHub\vcpkg\installed\x64-windows\include\qt5 -IG:\GitHub\vcpkg\installed\x64-windows\include\qt5\QtCore -IG:\GitHub\vcpkg\installed\x64-windows\tools\qt5\mkspecs\win32-msvc -IG:\GitHub\vcpkg\installed\x64-windows\include\qt5\QtOpenGL -IG:\GitHub\vcpkg\installed\x64-windows\include\qt5\QtWidgets -IG:\GitHub\vcpkg\installed\x64-windows\include\qt5\QtGui /nologo /DWIN32 /D_WINDOWS  /utf-8 /GR /EHsc /MP  /EHsc /MP -openmp /W0 /MD /O2 /Oi /Gy /DNDEBUG /Z7 /showIncludes /Folib\PoissonRecon\CMakeFiles\poisson_recon.dir\Factor.cpp.obj /Fdlib\PoissonRecon\CMakeFiles\poisson_recon.dir\poisson_recon.pdb /FS -c G:\GitHub\vcpkg\buildtrees\colmap\src\d576485fe6-7dd36362d1.clean\lib\PoissonRecon\Factor.cpp
...
Skipped 7021 lines
...

G:\GitHub\vcpkg\buildtrees\colmap\src\d576485fe6-7dd36362d1.clean\lib\FLANN/util/serialization.h(42): note: see reference to function template instantiation 'void flann::serialization::access::serialize<Archive,T>(Archive &,T &)' being compiled
        with
        [
            Archive=flann::serialization::LoadArchive,
            T=flann::KDTreeIndex<flann::L2<uint8_t>>::Node
        ]
G:\GitHub\vcpkg\buildtrees\colmap\src\d576485fe6-7dd36362d1.clean\lib\FLANN/util/serialization.h(51): note: see reference to function template instantiation 'void flann::serialization::serialize<InputArchive,T>(Archive &,T &)' being compiled
        with
        [
            InputArchive=flann::serialization::LoadArchive,
            T=flann::KDTreeIndex<flann::L2<uint8_t>>::Node,
            Archive=flann::serialization::LoadArchive
        ]
G:\GitHub\vcpkg\buildtrees\colmap\src\d576485fe6-7dd36362d1.clean\lib\FLANN/util/serialization.h(298): note: see reference to function template instantiation 'void flann::serialization::Serializer<T>::load<Archive>(InputArchive &,T &)' being compiled
        with
        [
            T=flann::KDTreeIndex<flann::L2<uint8_t>>::Node,
            Archive=flann::serialization::LoadArchive,
            InputArchive=flann::serialization::LoadArchive
        ]
G:\GitHub\vcpkg\buildtrees\colmap\src\d576485fe6-7dd36362d1.clean\lib\FLANN/util/serialization.h(298): note: see reference to function template instantiation 'void flann::serialization::Serializer<T>::load<Archive>(InputArchive &,T &)' being compiled
        with
        [
            T=flann::KDTreeIndex<flann::L2<uint8_t>>::Node,
            Archive=flann::serialization::LoadArchive,
            InputArchive=flann::serialization::LoadArchive
        ]
G:\GitHub\vcpkg\buildtrees\colmap\src\d576485fe6-7dd36362d1.clean\lib\FLANN/algorithms/kdtree_index.h(184): note: see reference to function template instantiation 'Archive &flann::serialization::InputArchive<Archive>::operator &<flann::KDTreeIndex<Distance>::Node>(T &)' being compiled
        with
        [
            Archive=flann::serialization::LoadArchive,
            Distance=flann::L2<uint8_t>,
            T=flann::KDTreeIndex<flann::L2<uint8_t>>::Node
        ]
G:\GitHub\vcpkg\buildtrees\colmap\src\d576485fe6-7dd36362d1.clean\lib\FLANN/algorithms/kdtree_index.h(184): note: see reference to function template instantiation 'Archive &flann::serialization::InputArchive<Archive>::operator &<flann::KDTreeIndex<Distance>::Node>(T &)' being compiled
        with
        [
            Archive=flann::serialization::LoadArchive,
            Distance=flann::L2<uint8_t>,
            T=flann::KDTreeIndex<flann::L2<uint8_t>>::Node
        ]
G:\GitHub\vcpkg\buildtrees\colmap\src\d576485fe6-7dd36362d1.clean\lib\FLANN/util/serialization.h(34): note: see reference to function template instantiation 'void flann::KDTreeIndex<Distance>::serialize<Archive>(Archive &)' being compiled
        with
        [
            Distance=flann::L2<uint8_t>,
            Archive=flann::serialization::LoadArchive
        ]
G:\GitHub\vcpkg\buildtrees\colmap\src\d576485fe6-7dd36362d1.clean\lib\FLANN/util/serialization.h(34): note: see reference to function template instantiation 'void flann::KDTreeIndex<Distance>::serialize<Archive>(Archive &)' being compiled
        with
        [
            Distance=flann::L2<uint8_t>,
            Archive=flann::serialization::LoadArchive
        ]
G:\GitHub\vcpkg\buildtrees\colmap\src\d576485fe6-7dd36362d1.clean\lib\FLANN/algorithms/kdtree_index.h(344): warning C4291: 'void *operator new(size_t,flann::PooledAllocator &)': no matching operator delete found; memory will not be freed if initialization throws an exception
G:\GitHub\vcpkg\buildtrees\colmap\src\d576485fe6-7dd36362d1.clean\lib\FLANN/util/allocator.h(197): note: see declaration of 'operator new'
G:\GitHub\vcpkg\buildtrees\colmap\src\d576485fe6-7dd36362d1.clean\lib\FLANN/algorithms/hierarchical_clustering_index.h(413): warning C4291: 'void *operator new(size_t,flann::PooledAllocator &)': no matching operator delete found; memory will not be freed if initialization throws an exception
G:\GitHub\vcpkg\buildtrees\colmap\src\d576485fe6-7dd36362d1.clean\lib\FLANN/util/allocator.h(197): note: see declaration of 'operator new'
G:\GitHub\vcpkg\buildtrees\colmap\src\d576485fe6-7dd36362d1.clean\lib\FLANN/util/serialization.h(34): note: see reference to function template instantiation 'void flann::HierarchicalClusteringIndex<flann::DummyDistance>::Node::serialize<Archive>(Archive &)' being compiled
        with
        [
            Archive=flann::serialization::LoadArchive
        ]
G:\GitHub\vcpkg\buildtrees\colmap\src\d576485fe6-7dd36362d1.clean\lib\FLANN/util/serialization.h(34): note: see reference to function template instantiation 'void flann::HierarchicalClusteringIndex<flann::DummyDistance>::Node::serialize<Archive>(Archive &)' being compiled
        with
        [
            Archive=flann::serialization::LoadArchive
        ]
G:\GitHub\vcpkg\buildtrees\colmap\src\d576485fe6-7dd36362d1.clean\lib\FLANN/util/serialization.h(42): note: see reference to function template instantiation 'void flann::serialization::access::serialize<Archive,T>(Archive &,T &)' being compiled
        with
        [
            Archive=flann::serialization::LoadArchive,
            T=flann::HierarchicalClusteringIndex<flann::DummyDistance>::Node
        ]
G:\GitHub\vcpkg\buildtrees\colmap\src\d576485fe6-7dd36362d1.clean\lib\FLANN/util/serialization.h(51): note: see reference to function template instantiation 'void flann::serialization::serialize<InputArchive,T>(Archive &,T &)' being compiled
        with
        [
            InputArchive=flann::serialization::LoadArchive,
            T=flann::HierarchicalClusteringIndex<flann::DummyDistance>::Node,
            Archive=flann::serialization::LoadArchive
        ]
G:\GitHub\vcpkg\buildtrees\colmap\src\d576485fe6-7dd36362d1.clean\lib\FLANN/util/serialization.h(298): note: see reference to function template instantiation 'void flann::serialization::Serializer<T>::load<Archive>(InputArchive &,T &)' being compiled
        with
        [
            T=flann::HierarchicalClusteringIndex<flann::DummyDistance>::Node,
            Archive=flann::serialization::LoadArchive,
            InputArchive=flann::serialization::LoadArchive
        ]
G:\GitHub\vcpkg\buildtrees\colmap\src\d576485fe6-7dd36362d1.clean\lib\FLANN/util/serialization.h(298): note: see reference to function template instantiation 'void flann::serialization::Serializer<T>::load<Archive>(InputArchive &,T &)' being compiled
        with
        [
            T=flann::HierarchicalClusteringIndex<flann::DummyDistance>::Node,
            Archive=flann::serialization::LoadArchive,
            InputArchive=flann::serialization::LoadArchive
        ]
G:\GitHub\vcpkg\buildtrees\colmap\src\d576485fe6-7dd36362d1.clean\lib\FLANN/algorithms/hierarchical_clustering_index.h(254): note: see reference to function template instantiation 'Archive &flann::serialization::InputArchive<Archive>::operator &<flann::HierarchicalClusteringIndex<flann::DummyDistance>::Node>(T &)' being compiled
        with
        [
            Archive=flann::serialization::LoadArchive,
            T=flann::HierarchicalClusteringIndex<flann::DummyDistance>::Node
        ]
G:\GitHub\vcpkg\buildtrees\colmap\src\d576485fe6-7dd36362d1.clean\lib\FLANN/algorithms/hierarchical_clustering_index.h(254): note: see reference to function template instantiation 'Archive &flann::serialization::InputArchive<Archive>::operator &<flann::HierarchicalClusteringIndex<flann::DummyDistance>::Node>(T &)' being compiled
        with
        [
            Archive=flann::serialization::LoadArchive,
            T=flann::HierarchicalClusteringIndex<flann::DummyDistance>::Node
        ]
G:\GitHub\vcpkg\buildtrees\colmap\src\d576485fe6-7dd36362d1.clean\lib\FLANN/util/serialization.h(34): note: see reference to function template instantiation 'void flann::HierarchicalClusteringIndex<flann::DummyDistance>::serialize<Archive>(Archive &)' being compiled
        with
        [
            Archive=flann::serialization::LoadArchive
        ]
G:\GitHub\vcpkg\buildtrees\colmap\src\d576485fe6-7dd36362d1.clean\lib\FLANN/util/serialization.h(34): note: see reference to function template instantiation 'void flann::HierarchicalClusteringIndex<flann::DummyDistance>::serialize<Archive>(Archive &)' being compiled
        with
        [
            Archive=flann::serialization::LoadArchive
        ]
G:\GitHub\vcpkg\installed\x64-windows\include\Eigen\src\Core\PlainObjectBase.h(591) : fatal error C1001: Internal compiler error.
(compiler file 'D:\a\_work\1\s\src\vctools\Compiler\Utc\src\p2\main.c', line 224)
 To work around this problem, try simplifying or changing the program near the locations listed above.
If possible please provide a repro here: https://developercommunity.visualstudio.com 
Please choose the Technical Support command on the Visual C++ 
 Help menu, or open the Technical Support help file for more information
  cl!RaiseException()+0x69
  cl!RaiseException()+0x69
  cl!InvokeCompilerPassW()+0x87d1b
  cl!InvokeCompilerPassW()+0xf2a27

[134/342] "D:\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.34.31933\bin\Hostx64\x64\cl.exe"   /TP -DBOOST_ALL_DYN_LINK=1 -DCGAL_ENABLED -DCGAL_USE_GMPXX=1 -DCUDA_ENABLED -DGFLAGS_IS_A_DLL=1 -DGLOG_NO_ABBREVIATED_SEVERITIES -DGL_GLEXT_PROTOTYPES -DGUI_ENABLED -DIMATH_DLL -DOPENEXR_DLL -DOPENGL_ENABLED -DOPENMP_ENABLED -DQT_CORE_LIB -DQT_GUI_LIB -DQT_NO_DEBUG -DQT_OPENGL_LIB -DQT_WIDGETS_LIB -DWIN32_LEAN_AND_MEAN -D_MWAITXINTRIN_H_INCLUDED -IG:\GitHub\vcpkg\buildtrees\colmap\src\d576485fe6-7dd36362d1.clean\lib -IG:\GitHub\vcpkg\buildtrees\colmap\src\d576485fe6-7dd36362d1.clean\src -I"C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.7\include" -external:IG:\GitHub\vcpkg\installed\x64-windows\include -external:IG:\GitHub\vcpkg\installed\x64-windows\include\eigen3 -external:IG:\GitHub\vcpkg\installed\x64-windows\include\qt5 -external:IG:\GitHub\vcpkg\installed\x64-windows\include\qt5\QtCore -external:IG:\GitHub\vcpkg\installed\x64-windows\tools\qt5\mkspecs\win32-msvc -external:IG:\GitHub\vcpkg\installed\x64-windows\include\qt5\QtOpenGL -external:IG:\GitHub\vcpkg\installed\x64-windows\include\qt5\QtWidgets -external:IG:\GitHub\vcpkg\installed\x64-windows\include\qt5\QtGui -external:IG:\GitHub\vcpkg\installed\x64-windows\include\webp -external:IG:\GitHub\vcpkg\installed\x64-windows\include\OpenEXR -external:IG:\GitHub\vcpkg\installed\x64-windows\include\Imath -external:IG:\GitHub\vcpkg\installed\x64-windows\include\GKlib -external:W0 /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /EHsc /MP -openmp /W3 /MD /O2 /Oi /Gy /DNDEBUG /Z7  /EHsc /Zc:__cplusplus -D_SCL_SECURE_NO_DEPRECATE -D_SCL_SECURE_NO_WARNINGS /fp:strict /fp:except- /bigobj /showIncludes /Fosrc\CMakeFiles\colmap.dir\estimators\generalized_relative_pose.cc.obj /Fdsrc\CMakeFiles\colmap.dir\colmap.pdb /FS -c G:\GitHub\vcpkg\buildtrees\colmap\src\d576485fe6-7dd36362d1.clean\src\estimators\generalized_relative_pose.cc
[135/342] "D:\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.34.31933\bin\Hostx64\x64\cl.exe"   /TP -DBOOST_ALL_DYN_LINK=1 -DCGAL_ENABLED -DCGAL_USE_GMPXX=1 -DCUDA_ENABLED -DGFLAGS_IS_A_DLL=1 -DGLOG_NO_ABBREVIATED_SEVERITIES -DGL_GLEXT_PROTOTYPES -DGUI_ENABLED -DIMATH_DLL -DOPENEXR_DLL -DOPENGL_ENABLED -DOPENMP_ENABLED -DQT_CORE_LIB -DQT_GUI_LIB -DQT_NO_DEBUG -DQT_OPENGL_LIB -DQT_WIDGETS_LIB -DWIN32_LEAN_AND_MEAN -D_MWAITXINTRIN_H_INCLUDED -IG:\GitHub\vcpkg\buildtrees\colmap\src\d576485fe6-7dd36362d1.clean\lib -IG:\GitHub\vcpkg\buildtrees\colmap\src\d576485fe6-7dd36362d1.clean\src -I"C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.7\include" -external:IG:\GitHub\vcpkg\installed\x64-windows\include -external:IG:\GitHub\vcpkg\installed\x64-windows\include\eigen3 -external:IG:\GitHub\vcpkg\installed\x64-windows\include\qt5 -external:IG:\GitHub\vcpkg\installed\x64-windows\include\qt5\QtCore -external:IG:\GitHub\vcpkg\installed\x64-windows\tools\qt5\mkspecs\win32-msvc -external:IG:\GitHub\vcpkg\installed\x64-windows\include\qt5\QtOpenGL -external:IG:\GitHub\vcpkg\installed\x64-windows\include\qt5\QtWidgets -external:IG:\GitHub\vcpkg\installed\x64-windows\include\qt5\QtGui -external:IG:\GitHub\vcpkg\installed\x64-windows\include\webp -external:IG:\GitHub\vcpkg\installed\x64-windows\include\OpenEXR -external:IG:\GitHub\vcpkg\installed\x64-windows\include\Imath -external:IG:\GitHub\vcpkg\installed\x64-windows\include\GKlib -external:W0 /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /EHsc /MP -openmp /W3 /MD /O2 /Oi /Gy /DNDEBUG /Z7  /EHsc /Zc:__cplusplus -D_SCL_SECURE_NO_DEPRECATE -D_SCL_SECURE_NO_WARNINGS /fp:strict /fp:except- /bigobj /showIncludes /Fosrc\CMakeFiles\colmap.dir\estimators\generalized_absolute_pose_coeffs.cc.obj /Fdsrc\CMakeFiles\colmap.dir\colmap.pdb /FS -c G:\GitHub\vcpkg\buildtrees\colmap\src\d576485fe6-7dd36362d1.clean\src\estimators\generalized_absolute_pose_coeffs.cc
ninja: build stopped: subcommand failed.
```
</details>

**Additional context**

Add any other context about the problem here, such as what you have already tried to resolve the issue.

**Comment by autoantwort** [comment]
> fatal error C1001: Internal compiler error.

I think there is nothing vcpkg can do about that.

**Comment by LilyWangLL** [comment]
This is a Visual Studio bug, it has been fixed, need waiting for Visual Studio 17.6 Preview1 released. You could use Visual Studio 2019 or Visual Studio 17.6 Preview1 to install `colmap` temporarily.
